### PR TITLE
Simple bootloader hint 3: divide nums by 2

### PR DESCRIPTION
**Pull request by odesenfans**
## Checklist
- [ ] Linked to Github Issue
- [x] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [ ] CHANGELOG has been updated.

### CODE REVIEW

**Comment by odesenfans** [comment]
Depends on #13.

File: vm/src/hint_processor/builtin_hint_processor/bootloader/simple_bootloader_hints.rs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+/// %{ ids.num // 2 %}
+pub fn divide_num_by_2(
+    vm: &mut VirtualMachine,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    let felt = get_integer_from_var_name("num", vm, ids_data, ap_tracking)?.into_owned();
+
+    let num = felt
+        .to_u64()
```
Comment (notlesh) [inline vm/src/hint_processor/builtin_hint_processor/bootloader/simple_bootloader_hints.rs:76]: Is it clear that `u64` is sufficient? I don't think it is if it comes from here: https://github.com/lambdaclass/cairo-lang/blob/02d511208cece64da1300445ab0586cf75205e70/src/starkware/cairo/bootloaders/simple_bootloader/run_simple_bootloader.cairo#L121

If so, I think a `U256` is appropriate.

File: vm/src/hint_processor/builtin_hint_processor/bootloader/simple_bootloader_hints.rs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        .ok_or(HintError::Math(MathError::Felt252ToU64Conversion(
+            Box::new(felt),
+        )))?;
+
+    let num_divided_by_2 = num / 2;
```
Comment (notlesh) [inline vm/src/hint_processor/builtin_hint_processor/bootloader/simple_bootloader_hints.rs:81]: Should this use `field division` defined here? https://cairo-by-example.com/examples/field_arithmetic/

**Review by notlesh — CHANGES_REQUESTED** [review]
Left a couple things that could use a close look

File: vm/src/hint_processor/builtin_hint_processor/bootloader/simple_bootloader_hints.rs (hunk 1 above)
Comment (odesenfans) [reply vm/src/hint_processor/builtin_hint_processor/bootloader/simple_bootloader_hints.rs:76]: Yeah you're right, I missed the context of where this is used.

File: vm/src/hint_processor/builtin_hint_processor/bootloader/simple_bootloader_hints.rs (hunk 2 above)
Comment (odesenfans) [reply vm/src/hint_processor/builtin_hint_processor/bootloader/simple_bootloader_hints.rs:81]: That's an excellent question. The thing that throws me off is that this hint uses floor division (`//` operator).

**Review by notlesh — APPROVED** [review]

**Pull request merged**